be/src/runtime/workload_group/workload_group.h

```diff
@@ -141,7 +166,8 @@ class WorkloadGroup : public std::enable_shared_from_this<WorkloadGroup> {
     const uint64_t _id;
     std::string _name;
     int64_t _version;
-    int64_t _memory_limit; // bytes
+    int64_t _memory_limit;                      // bytes
+    std::atomic_int64_t _weighted_mem_used = 0; // bytes
     bool _enable_memory_overcommit;
     std::atomic<uint64_t> _cpu_share;
```

Comment:
remove query_id_set and use std::map<queryid, weak_ptr<query context>> so that all memory limit info could be set to query context, not to memtracker.